cpp/inc/bond/protocol/simple_json_reader.h

```diff
@@ -108,6 +108,13 @@ class SimpleJsonReader
     {
         return _value == rhs._value;
     }
+
+    /// @brief Access to underlaying buffer
+    typename boost::call_traits<Buffer>::reference
```

Comment:
Add a const overload as well, to match the other protocols? Similalrly for simple_json_writer.